plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -68,3 +69,69 @@ export type ReadLocationError = {
   location: LocationSpec;
   error: Error;
 };
+
+//
+// LocationAnalyzer
+//
+
+export type LocationAnalyzer = {
+  generateConfig(
+    location: AnalyzeLocationRequest,
+  ): Promise<AnalyzeLocationResponse>;
+};
+
+export type AnalyzeLocationRequest = {
+  location: LocationSpec;
+};
+
+export type AnalyzeLocationResponse = {
+  existingEntityFiles: AnalyzeLocationExistingEntity[];
+  generateEntities: AnalyzeLocationGenerateEntity[];
+};
+
+// If the folder pointed to already contained catalog info yaml files, they are
+// read and emitted like this so that the frontend can inform the user that it
+// located them and can make sure to register them as well if they weren't
+// already
+type AnalyzeLocationExistingEntity = {
+  location: LocationSpec;
+  isRegistered: boolean;
+  entity: Entity;
+};
+
+// This is some form of representation of what the analyzer could deduce.
+// We should probably have a chat about how this can best be conveyed to
+// the frontend. It'll probably contain a (possibly incomplete) entity, plus
+// enough info for the frontend to know what form data to show to the user
+// for overriding/completing the info.
```

Comment:
As @freben has mentioned in discussions it could make sense to transform this into JSON Schema, with existing values, missing fields, etc. I'd really like to keep that kind of logic outside of the catalog backend itself though, keeping the additions for this use-case to the bare minimum needed to support it and instead having a separate `catalog-import-backend` plugin or so that does most of the work.

My hunch here is that it would essentially be enough to be able to dry-run catalog processing without the final validation step, and then use the output entities and the base for that schema by inspecting them for missing fields etc.